### PR TITLE
apis: short names for common resources

### PR DESCRIPTION
**Pull request by nicks**
Hello @landism, @milas,

Please review the following commits I made in branch nicks/shortnames:

9bbbf3cdbdef01508b251ad5b70334e0572356ab (2021-07-07 16:18:38 -0400)
apis: short names for common resources

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
following the k8s convention of having short two-letter names, and more human-friendly names. here are the current api-resources:

```
$ tilt api-resources
NAME                    SHORTNAMES   APIVERSION          NAMESPACED   KIND
cmds                                 tilt.dev/v1alpha1   false        Cmd
filewatches             fw           tilt.dev/v1alpha1   false        FileWatch
imagemaps               im           tilt.dev/v1alpha1   false        ImageMap
kubernetesapplys        ka,kapp      tilt.dev/v1alpha1   false        KubernetesApply
kubernetesdiscoveries   kd,kdisco    tilt.dev/v1alpha1   false        KubernetesDiscovery
podlogstreams           pls          tilt.dev/v1alpha1   false        PodLogStream
portforwards            pf           tilt.dev/v1alpha1   false        PortForward
sessions                             tilt.dev/v1alpha1   false        Session
uibuttons                            tilt.dev/v1alpha1   false        UIButton
uiresources                          tilt.dev/v1alpha1   false        UIResource
uisessions                           tilt.dev/v1alpha1   false        UISession
```

File: pkg/apis/core/v1alpha1/kubernetesdiscovery_types.go [hunk 1]
```diff
@@ -152,6 +154,10 @@ func (in *KubernetesDiscovery) NamespaceScoped() bool {
 	return false
 }
 
+func (in *KubernetesDiscovery) ShortNames() []string {
+	return []string{"kd", "kdisco"}
```
Comment (landism) [inline pkg/apis/core/v1alpha1/kubernetesdiscovery_types.go:158]: 🤔 emoji are legal here:
```
NAME                    SHORTNAMES    APIVERSION          NAMESPACED   KIND
cmds                                  tilt.dev/v1alpha1   false        Cmd
filewatches             fw            tilt.dev/v1alpha1   false        FileWatch
imagemaps                             tilt.dev/v1alpha1   false        ImageMap
kubernetesapplys                      tilt.dev/v1alpha1   false        KubernetesApply
kubernetesdiscoveries   kd,kdisco,🕺   tilt.dev/v1alpha1   false        KubernetesDiscovery
```

**Review by landism — APPROVED** [review]
🙌

**Comment by nicks** [comment]
:joy:

**Pull request merged**